libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -394,13 +393,13 @@ internal class MapboxTripSession(
         unconditionalStatusPollingJob?.cancel()
         this.rawLocation = rawLocation
         locationObservers.forEach { it.onRawLocationChanged(rawLocation) }
-        navigatorJobController.scope.launch {
+        ioJobController.scope.launch {
```

Comment:
I think we're doing too much thread jumping here. We're dispatching on the IO thread in this line, just to switch to another IO thread in the `navigator` impl. Can we remove this thread switch and dispatch a main thread coroutine if we're already switching to the IO thread in the `navigator` impl? This would simplify `MapboxTripSession` a lot.

Reply:
As noted in https://github.com/mapbox/mapbox-navigation-android/issues/3463#issuecomment-676551365 there are some implementation details challenges still to be thought 

> From which dispatcher should we run these `suspend`ed functions? There is code that needs to be executed out of `Main`

So if I understood you correctly you're specifically referring to `Main.Dispatcher` > `IODispatcher` > `NavigatorDispatcher` situations, right? If so, yeah we can save some jumps. Was thinking if there was another solution to tackle https://github.com/mapbox/mapbox-navigation-android/issues/3463#issue-681497066

> We could do this by following a "withContext(NavigatorDispatcher)" pattern for all/most interactions with native.

BTW dispatcher switch is instantaneous, noticed / learned when working / testing https://github.com/mapbox/mapbox-navigation-android/pull/2719 https://github.com/mapbox/mapbox-navigation-android/issues/3265 https://github.com/mapbox/mapbox-navigation-android/pull/3255/files#r447617327 so it shouldn't be an issue after all.

Reply:
Remove the unnecessary `MapboxTripSession` thread switching in https://github.com/mapbox/mapbox-navigation-android/pull/3471/commits/0b74267b7ffd90ea63bfdd646ce0d1e014a40592 Let me know if that's what you were thinking and if it looks good to you 🙏